library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -209,6 +229,61 @@ class Client() {
         publishUserContact(legacy = true)
     }
 
+    fun importConversation(conversationData: ByteArray): Conversation {
+        val gson = GsonBuilder().create()
+        val v2Export = gson.fromJson(
+            conversationData.toString(StandardCharsets.UTF_8),
+            ConversationV2Export::class.java
+        )
+        try {
+            return importV2Conversation(export = v2Export)
+        } catch (e: java.lang.Exception) {
+            val v1Export = gson.fromJson(
+                conversationData.toString(StandardCharsets.UTF_8),
+                ConversationV1Export::class.java
+            )
+            try {
+                return importV1Conversation(export = v1Export)
+            } catch (e: java.lang.Exception) {
+                throw XMTPException("Invalid input data", e)
+            }
+        }
+    }
+
+    fun importV2Conversation(export: ConversationV2Export): Conversation {
+        val keyMaterial = Base64.decode(export.keyMaterial)
+        return Conversation.V2(
+            ConversationV2(
+                topic = export.topic,
+                keyMaterial = keyMaterial,
+                context = InvitationV1ContextBuilder.buildFromConversation(
+                    conversationId = export.context?.conversationId ?: "",
+                    metadata = export.context?.metadata ?: mapOf()
+                ),
+                peerAddress = export.peerAddress,
+                client = this,
+                header = SealedInvitationHeaderV1.newBuilder().build()
+            )
+        )
+    }
+
+    fun importV1Conversation(export: ConversationV1Export): Conversation {
+        val sentAt = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            Date.from(Instant.parse(export.createdAt))
+        } else {
+            val df = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.getDefault())
+            df.timeZone = TimeZone.getTimeZone("UTC")
+            df.parse(export.createdAt)
```

Comment:
@elisealix22 I found something that works with version 22. SimpleDateFormat is not ideal since if anything differently formatted gets returned this will break so I'm keeping these still split into a sdk version check. But now the date will be accurate.